src/khoj/routers/api.py

```diff
@@ -410,13 +435,10 @@ async def search(
         defiltered_query = filter.defilter(defiltered_query)
 
     encoded_asymmetric_query = None
-    if t == SearchType.All or t != SearchType.Image:
-        text_search_models: List[TextSearchModel] = [
-            model for model in state.search_models.__dict__.values() if isinstance(model, TextSearchModel)
-        ]
-        if text_search_models:
-            with timer("Encoding query took", logger=logger):
-                encoded_asymmetric_query = state.embeddings_model.embed_query(defiltered_query)
+    if t != SearchType.Image:
```

Comment:
Don't we still need to set the `encoded_asymmetric_query` when `t == SearchType.All`?

Reply:
That should still be true with the removed condition, but yes.

Reply:
Ah, I see. Makes sense